shared/net/addrs.go

```diff
@@ -141,23 +157,21 @@ func GetIPAddresses(ifaces ...string) ([]netip.Addr, error) {
 			return out, err
 		}
 
-		for _, addr := range addrs {
-			var s []byte
+		out = appendNetIPAsIP(out, addrs...)
+	}
 
-			switch v := addr.(type) {
-			case *net.IPAddr:
-				s = v.IP
-			case *net.IPNet:
-				s = v.IP
-			}
+	return out, nil
+}
 
-			if ip, ok := netip.AddrFromSlice(s); ok {
-				out = append(out, ip)
-			}
+func appendNetIPAsIP(out []netip.Addr, addrs ...net.Addr) []netip.Addr {
+	for _, netAddr := range addrs {
+		addr, err := netip.ParseAddr(netAddr.String())
```

Comment:
this line has a "bug" on Linux. On Linux, as opposed to Windows, `net.InterfaceAddrs()` will return addresses with CIDR notation. netip.ParseAddr is chocking on the CIDR notation and returns invalid address.
the fix could be:
```go
func appendNetIPAsIP(out []netip.Addr, addrs ...net.Addr) []netip.Addr {
	for _, netAddr := range addrs {
		netAddrStr := strings.Split(netAddr.String(), "/")[0]
		addr, err := netip.ParseAddr(netAddrStr)
		if err == nil && addr.IsValid() {
			out = append(out, addr)
		}
	}

	return out
}
```

Reply:
Or even an one-liner:
```go
addr, err := netip.ParseAddr(strings.Split(netAddr.String(), "/")[0])
```